src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -364,87 +364,7 @@ public override void Abort()
-
-            // If we get here, the cancellation token is not cancelable so we don't have to worry about it,
-            // and we own the semaphore, so we don't need to asynchronously wait for it.
-            Task writeTask = null;
-            bool releaseSemaphoreAndSendBuffer = true;
-            try
-            {
-                // Write the payload synchronously to the buffer, then write that buffer out to the network.
-                int sendBytes = WriteFrameToSendBuffer(opcode, endOfMessage, payloadBuffer.Span);
-                writeTask = _stream.WriteAsync(_sendBuffer, 0, sendBytes, CancellationToken.None);
-
-                // If the operation happens to complete synchronously (or, more specifically, by
-                // the time we get from the previous line to here), release the semaphore, return
-                // the task, and we're done.
-                if (writeTask.IsCompleted)
-                {
-                    return writeTask;
-                }
-
-                // Up until this point, if an exception occurred (such as when accessing _stream or when
-                // calling GetResult), we want to release the semaphore and the send buffer. After this point,
-                // both need to be held until writeTask completes.
-                releaseSemaphoreAndSendBuffer = false;
-            }
-            catch (Exception exc)
-            {
-                return Task.FromException(
-                    exc is OperationCanceledException ? exc :
-                    _state == WebSocketState.Aborted ? CreateOperationCanceledException(exc) :
-                    new WebSocketException(WebSocketError.ConnectionClosedPrematurely, exc));
-            }
-            finally
-            {
-                if (releaseSemaphoreAndSendBuffer)
-                {
-                    _sendFrameAsyncLock.Release();
-                    ReleaseSendBuffer();
-                }
-            }
-
-            // The write was not yet completed.  Create and return a continuation that will
-            // release the semaphore and translate any exception that occurred.
-            return writeTask.ContinueWith((t, s) =>
-            {
-                var thisRef = (ManagedWebSocket)s;
-                thisRef._sendFrameAsyncLock.Release();
-                thisRef.ReleaseSendBuffer();
-
-                try { t.GetAwaiter().GetResult(); }
-                catch (Exception exc) when (!(exc is OperationCanceledException))
-                {
-                    throw thisRef._state == WebSocketState.Aborted ?
-                        CreateOperationCanceledException(exc) :
-                        new WebSocketException(WebSocketError.ConnectionClosedPrematurely, exc);
-                }
-            }, this, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+            return SendFrameFallbackAsync(opcode, endOfMessage, payloadBuffer, cancellationToken);
```

Comment:
Can you revert this piece?  I've actually addressed it in https://github.com/dotnet/corefx/pull/27497, mainly by removing the comment.  When I initially wrote this, I thought we could re-simplify this after #4900, but it turns out there are other benefits to keeping things the way I had them.

Reply:
Yep, I can do that!